OSDOCS-13291:RN for UDN GA (#88105)
openshift/openshift-docs: release_notes/ocp-4-18-release-notes.adoc

* `NetworkPolicy`
* `AdminNetworkPolicy`
* `BaselineNetworkPolicy`
* `UserDefinedNetwork` isolation
* `EgressFirewall`
* Multicast ACLs.

Checking OVN-Kubernetes network traffic with OVS sampling using the CLI is intended to help with packet tracing. It can also be used while the Network Observability Operator is installed.

For more information, see xref:../networking/ovn_kubernetes_network_provider/ovn-kubernetes-troubleshooting-sources.adoc#nw-ovn-kubernetes-observability_ovn-kubernetes-sources-of-troubleshooting-information[Checking OVN-Kubernetes network traffic with OVS sampling using the CLI].

[id="ocp-release-notes-networking-user-defined-network-segmentation_{context}"]
==== User-defined network segmentation (Generally Available)

> With {product-title} {product-y}, user-defined network segmentation is generally available. User-defined networks (UDN) provide cluster administrators and users with highly customizable network configuration options for both primary and secondary network types. User-defined network segmentation is comprised of a namespace-scoped `UserDefinedNetwork` custom resource and a cluster-scoped `ClusterUserDefinedNetwork` custom resource.

Reviewer: ```suggestion
options for both primary and secondary network types. 
```
on a first glance this doesn't look appealing to someone reading it because I am sure no one knows what's primary and secondary here. I like what you have in 2nd para around UDNs can be used to segment your workloads and isolate them natively - that's an easier way to put it. I'd say dumb it down a bit more in layperson terms..

Reviewer 2: I agree with the "primary/secondary" feedback that it will be confusing to customers, but maybe for a different reason -- they might think that we're referring to Multus-enabled add-on CNI plug-in primary/secondary networks.  A reader that doesn't know the difference between primary/secondary networks is likely not the target audience for the feature (so I'm not sure we need to explain what they are?), but separating UDN primary/secondary from Multus primary/secondary should be clear cut. 

If you think we still need layperson for primary/secondary, its fine to add a brief explanation (however you see fit) that primary is where Kubernetes control plane traffic traffic travels, and is where traffic flows by default (default gateway/route).   Secondary interfaces are used to segment traffic onto a completely different, isolated pod/VM network interface for whatever reason (primarily security, but also isolation from control plane for performance and so you don't disrupt control plane traffic from flowing freely).  UDNs do it differently, and better. 

For the UDN/Multus issue, maybe working in an additional sentence, something along the lines of: "UDN is different than, but augments OpenShift’s Multus-enabled secondary CNI capability by providing a comparable experience and feature set to all network segments, but with additional feature capability advantages."  ?

Reviewer: yeah I agree with you, what I meant was we shouldn't have to even mention "primary/secondary" in the Release notes since the feature docs have it - but go with more around "what use cases this feature solves" which is what might attract the audience.
+1 to what you said - it might confuse with existing multus as well which i hadn't thought of before.. so maybe that part around primary/secondary can be striked here...

Reviewer 2: Agree. 

For "what use cases this feature solves", I've gotten similar feedback, and approached it a couple of ways, so far, stated here purely to incite ideas:

1. See draft text of the "What's New in 4.18" blog:  https://docs.google.com/document/d/1Wkkx7ws9_smm8RRqhVJwyogZSWDPJxTmeUUrlcp-Xz4/edit?tab=t.0
2. See the draft text of a UDN-specific blog (the last section, "Enter User Defined Networks"):  https://docs.google.com/document/d/1ofehoVU-iyI4CNkr5Pke6EJRW4cQloVxzZ148mX1bAw/edit?tab=t.0

I'm totally open to a better way to do it.

> With UDNs, administrators can create tailored network topologies with enhanced isolation, IP address management for workloads, and advanced networking features. xr3f:../networking/multiple_networks/understanding-multiple-networks.adoc#nw-udn-support-matrix-primary-secondary[Supporting] both Layer 2 and Layer 3 topology types, user-defined network segmentation enables a wide range of network architectures and topologies, enhancing network flexibility, security, and performance. For more information, see xref:../networking/multiple_networks/about-user-defined-networks.adoc#about-user-defined-networks[About user-defined networks].

Reviewer 3: On the topic of performance, we might just say "and more" here: 

```suggestion
With UDNs, administrators can create tailored network topologies with enhanced isolation, IP address management for workloads, and advanced networking features. xr3f:../networking/multiple_networks/understanding-multiple-networks.adoc#nw-udn-support-matrix-primary-secondary[Supporting] both Layer 2 and Layer 3 topology types, user-defined network segmentation enables a wide range of network architectures and topologies, enhancing network flexibility, security, and more. For more information, see xref:../networking/multiple_networks/about-user-defined-networks.adoc#about-user-defined-networks[About user-defined networks].
```

* `NetworkPolicy`
* `AdminNetworkPolicy`
* `BaselineNetworkPolicy`
* `UserDefinedNetwork` isolation
* `EgressFirewall`
* Multicast ACLs.

Checking OVN-Kubernetes network traffic with OVS sampling using the CLI is intended to help with packet tracing. It can also be used while the Network Observability Operator is installed.

For more information, see xref:../networking/ovn_kubernetes_network_provider/ovn-kubernetes-troubleshooting-sources.adoc#nw-ovn-kubernetes-observability_ovn-kubernetes-sources-of-troubleshooting-information[Checking OVN-Kubernetes network traffic with OVS sampling using the CLI].

[id="ocp-release-notes-networking-user-defined-network-segmentation_{context}"]
==== User-defined network segmentation (Generally Available)

> With {product-title} {product-version}, user-defined network segmentation is generally available. User-defined networks (UDN) introduce enhanced network segmentation capabilities by allowing administrators to define custom network topologies using namespace-scoped UserDefinedNetwork and cluster-scoped ClusterUserDefinedNetwork custom resources.

Reviewer 3: ```suggestion
With {product-title} {product-version}, user-defined network segmentation is generally available. User-defined networks (UDN) introduce enhanced network segmentation capabilities by allowing administrators to define custom network topologies using namespace-scoped `UserDefinedNetwork` and cluster-scoped `ClusterUserDefinedNetwork` custom resources.
```

Author: removed

With UDNs, administrators can create tailored network topologies with enhanced isolation, IP address management for workloads, and advanced networking features. Supporting both Layer 2 and Layer 3 topology types, user-defined network segmentation enables a wide range of network architectures and topologies, enhancing network flexibility, security, and performance. For more information on supported features, see xr3f:../../networking/multiple_networks/understanding-multiple-networks.adoc#nw-udn-support-matrix-primary-secondary[UDN support matrix].

> UDN's use cases include providing virtual machines (VMs) with a lifetime duration for static IP addresses assignment as well as a Layer 2 primary pod network so that users can live migrate VMs between nodes. These features are all fully equipped in {VirtProductName}. Users can use UDNs to create a stronger, native multi-tenant environment, allowing you to secure your overlay Kubernetes network, which is otherwise open by default. For more information, see xref:../networking/multiple_networks/primary_networks/about-user-defined-networks.adoc#about-user-defined-networks[About user-defined networks].

Reviewer 4: Suggestion to consider writing as "Use cases of...." instead of using UDN's

https://www.ibm.com/docs/en/ibm-style?topic=grammar-possessives

> Do not use possessive ’s with inanimate objects.

> UDN's use cases include providing virtual machines (VMs) with a lifetime duration for static IP addresses assignment as well as a layer 2 primary pod network so that users can live migrate VMs between nodes. These features are all fully equipped in {VirtProductName}. UDN's use case for network segmentation provides stronger multi-tenant environments by removing the need for the `AdminNetworkPolicy` resource while supporting fine grained micro-segmentation through network policy. For more information, see xref:../networking/multiple_networks/primary_networks/about-user-defined-networks.adoc#about-user-defined-networks[About user-defined networks].

Reviewer 3: Should we capitalize Layer 2/Layer 3 or not? it's not here, but is in the second paragraph. Also, sorry, but one more nit: 

". . . for the `AdminNetworkPolicy` custom resource" (note addition of custom.

Author: We do in the main docs.

> UDN's use cases include providing virtual machines (VMs) with a lifetime duration for static IP addresses assignment as well as a layer 2 primary pod network so that users can live migrate VMs between nodes. These features are all fully equipped in {VirtProductName}. UDN's use case for network segmentation provides stronger multi-tenant environments by removing the need for the `AdminNetworkPolicy` resource while supporting fine grained micro-segmentation through network policy. For more information, see xref:../networking/multiple_networks/primary_networks/about-user-defined-networks.adoc#about-user-defined-networks[About user-defined networks].

Reviewer: ```suggestion
UDN's use cases include providing virtual machines (VMs) with a lifetime duration for static IP addresses assignment as well as a layer 2 primary pod network so that users can live migrate VMs between nodes. These features are all fully equipped in {VirtProductName}. UDNs create a stronger, native multi-tenant environment, allowing users to secure their overlay Kubernetes network, which is otherwise open by default. For more information, see xref:../networking/multiple_networks/primary_networks/about-user-defined-networks.adoc#about-user-defined-networks[About user-defined networks].
```
I don't think UDNs remove the need for ANPs.. UDNs, ANPs and NPs are all supposed to be layered ways of achieving segmentation. Otherwise we'd not need to implement ANPs on UDNs which we are :) - ANPs are also going to allow admins for finer microsegmentation just like netpol. I'd avoid mentioning ANP or NetPol in RN but instead have a section in the UDN docs calling out that aspect
